### DISCUSSION TITLE
✨ Option to ignore hosts `.docker/config`

### DISCUSSION THREAD

**Discussion by kpenfound** [dagger/dagger#12418] [Ideas]
### What are you trying to do?

The current behavior is to load the hosts `.docker/config` into the dagger session so that existing registry authentications can be used. While this is generally a convenience, there are cases where this is not desirable and there should be an option to opt out of this behavior

**Comment by jedevc** [comment]
Isn't it also possible to `Publish` an image using this token? That maybe also feels like a bit of an issue, if we could do that.

**Comment by migmartri** [comment]
Thanks for reporting this issue.

This was, in fact, the root cause of an issue I ran into. 

A Dagger function that pulled a `public` container image stopped working just because, in my host, I had some invalid credentials (expired Github GCR credentials).

The error I got:

```
status from GET request to https://ghcr.io/token?scope=repository%3Achainloop-dev%2Fchainloop%2Fcli%3Apull&service=ghcr.io: 403 Forbidden
```

This was unexpected because my understanding was that dagger functions limited the environment information (i.e., Env Vars) injected automatically from the host. I was not expecting the docker context to be injected automatically.

Ideally, to prevent this issue from happening again, in my module, I'd like to force no authentication in my `dag.Container.From` chain.

I tried `WithoutRegistryAuth,` but it didn't seem to work, or I don't know how to use it.
